streams/src/main/java/org/apache/kafka/streams/processor/internals/ProcessorStateManager.java

```diff
@@ -509,11 +509,12 @@ void transitionTaskType(final TaskType newType, final LogContext logContext) {
             throw new IllegalStateException("Tried to recycle state for task type conversion but new type was the same.");
         }
 
+        TaskType oldType = taskType;
         taskType = newType;
         log = logContext.logger(ProcessorStateManager.class);
         logPrefix = logContext.logPrefix();
 
-        log.debug("Transitioning state manager for {} task {} to {}", taskType, taskId, newType);
+        log.debug("Transitioning state manager for {} task {} to {}", oldType, taskId, newType);
```

Comment:
Another alternative for the log message would be

```
standby-task [0_2] Transitioning state manager for STANDBY task 0_2 to ACTIVE 
```

Reply:
I agree it would be useful to prefix the logs with `standby` or `active`, but I'd prefer to do that everywhere in a separate PR.

Can we just move the log message to before we reassign `taskType = newType`? Or do you think we might forget/not notice that the ordering is relevant and accidentally move it back at some future time?

Reply:
I think using oldType/newType is fine here since then we do not need to keep in mind of the ordering of each call?